blueocean-pipeline-api-impl/src/main/java/io/jenkins/blueocean/rest/impl/pipeline/PipelineNodeImpl.java

```diff
@@ -152,16 +170,83 @@ public Link getLink() {
                                                   this);
     }
 
+    @Override
+    public boolean isRestartable()
+    {
+        RestartDeclarativePipelineAction restartDeclarativePipelineAction =
+            this.run.getAction( RestartDeclarativePipelineAction.class );
+        if ( restartDeclarativePipelineAction != null )
+        {
+            List<String> restartableStages = restartDeclarativePipelineAction.getRestartableStages();
+            if ( restartableStages != null )
+            {
+                return restartableStages.contains(this.getDisplayName())
```

Comment:
Should it be based on displayName or id? displayName is not unique AFAIK. cc: @abayer

Reply:
Looking at https://github.com/jenkinsci/pipeline-model-definition-plugin/blob/master/pipeline-model-definition/src/main/java/org/jenkinsci/plugins/pipeline/modeldefinition/actions/RestartDeclarativePipelineAction.java#L138 
Not sure we have the id...